src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -309,6 +309,38 @@ def _validate_location(cmd, namespace, zone_info, size_info):
                                "used to find such locations".format(namespace.resource_group_name))
 
 
+def _validate_vm_encryption_settings(cmd, namespace):
+    # if there is no disk_encryption* parameter then none of the encryption params should be provided
+    required = []
+    forbidden = ['disk_encryption_key', 'disk_encryption_keyvault', 'key_encryption_key', 'key_encryption_keyvault']
+
+    # if disk_encryption_key or disk_encryption_keyvault are present. They should both be required.
+    if getattr(namespace, 'disk_encryption_key', None) or getattr(namespace, 'disk_encryption_keyvault', None):
+        required = ['disk_encryption_key', 'disk_encryption_keyvault']
+        forbidden = ['key_encryption_key', 'key_encryption_keyvault']
```

Comment:
i thought kek is optional, could you please clarify?

Reply:
Yes KEK is optional.
The code below just makes sure that if one of kek vault or kek url is specified, the other is also specified.